fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/BaseWhereAdapter.java

```diff
@@ -0,0 +1,1137 @@
+import com.ibm.fhir.database.utils.query.node.CosExpNode;
+import com.ibm.fhir.database.utils.query.node.DoubleBindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.ExistsExpNode;
+import com.ibm.fhir.database.utils.query.node.ExpNode;
+import com.ibm.fhir.database.utils.query.node.InstantBindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.IntegerBindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.LongBindMarkerNode;
+import com.ibm.fhir.database.utils.query.node.NotExistsExpNode;
+import com.ibm.fhir.database.utils.query.node.PredicateParser;
+import com.ibm.fhir.database.utils.query.node.SelectExpNode;
+import com.ibm.fhir.database.utils.query.node.SinExpNode;
+import com.ibm.fhir.database.utils.query.node.StringBindMarkerNode;
+
+/**
+ * A piece of a where clause which isn't attached to a {@link Select} statement
+ * allowing getThis() bit to be reused where the statement/whereClause isn't
+ * yet established.
+ */
+public abstract class BaseWhereAdapter<T> {
+
+    // The parser used to build the where statement
+    private final PredicateParser predicateParser;
+
+    /**
+     * Returns the sub-class type-specific instance of the class. Used as
+     * the return value in fluent methods.
+     * @return
+     */
+    protected abstract T getThis();
+
+    /**
+     * Getter for the predicate parser
+     * @return
+     */
+    protected PredicateParser getPredicateParser() {
+        return this.predicateParser;
+    }
+
+    /**
+     * For building standalone predicates like filters
+     */
+    protected BaseWhereAdapter() {
+        this.predicateParser = new PredicateParser();
+    }
+
+    /**
+     * Adopts an existing predicate parser
+     * @param pp
+     */
+    protected BaseWhereAdapter(PredicateParser pp) {
+        this.predicateParser = pp;
+    }
+
+    /**
+     * Get the expression that has been built
+     * @return
+     */
+    public ExpNode getExpression() {
+        return this.predicateParser.parse();
+    }
+
+    // ### LITERALS ###
+    /**
+     * Add a literal String value to the expression
+     * @param value
+     * @return
+     */
+    public T literal(String value) {
+        predicateParser.literal(value);
+        return getThis();
+    }
+
+    /**
+     * Add a literal long value to the expression
+     * @param value
+     * @return
+     */
+    public T literal(long value) {
+        predicateParser.literal(value);
+        return getThis();
+    }
+
+    /**
+     * Add a literal double value to the expression
+     * @param value
+     * @return
+     */
+    public T literal(double value) {
+        predicateParser.literal(value);
+        return getThis();
+    }
+
+    // ### EQUAL ###
+    /**
+     * Add an equals '=' node to the expression
+     * @return
+     */
+    public T eq() {
+        predicateParser.eq();
+        return getThis();
+    }
+
+    /**
+     * Add '= <ref>' to the expression
```

Comment:
just so I follow... <ref> is meant to indicate that the ref parameter is inserted

Reply:
also about half way through this class it flips the style to use { } which I think is preferrable

Reply:
yeah, I realized that <xx> would probably upset some formatting but forgot to switch the old comments.

Reply:
Yes, it adds an EQUALS node, followed by a node representing the `ref` arg. Convenience function.

Reply:
now uses { } consistently